virtual_rainforest/models/abiotic_simple/abiotic_simple_model.py

```diff
@@ -0,0 +1,208 @@
+            raise to_raise
+
+        if soil_layers != int(soil_layers):
+            to_raise = InitialisationError(
+                "The number of soil layers must be an integer!"
+            )
+            LOGGER.error(to_raise)
+            raise to_raise
+
+        if canopy_layers < 1:
+            to_raise = InitialisationError(
+                "There has to be at least one canopy layer in the abiotic model!"
+            )
+            LOGGER.error(to_raise)
+            raise to_raise
+
+        if canopy_layers != int(canopy_layers):
+            to_raise = InitialisationError(
+                "The number of canopy layers must be an integer!"
+            )
+            LOGGER.error(to_raise)
+            raise to_raise
+
+        super().__init__(data, update_interval, **kwargs)
+
+        # create a list of layer roles
+        layer_roles = set_layer_roles(canopy_layers, soil_layers)
+
+        self.data
+        """A Data instance providing access to the shared simulation data."""
+        self.layer_roles = layer_roles
+        "A list of vertical layer roles."
+        self.update_interval
+        """The time interval between model updates."""
+
+    @classmethod
+    def from_config(cls, data: Data, config: dict[str, Any]) -> AbioticSimpleModel:
+        """Factory function to initialise the simple abiotic model from configuration.
+
+        This function unpacks the relevant information from the configuration file, and
+        then uses it to initialise the model. If any information from the config is
+        invalid rather than returning an initialised model instance an error is raised.
+
+        Args:
+            data: A :class:`~virtual_rainforest.core.data.Data` instance.
+            config: The complete (and validated) Virtual Rainforest configuration.
+        """
+
+        # Find timing details
+        update_interval = extract_update_interval(
+            config, cls.lower_bound_on_time_scale, cls.upper_bound_on_time_scale
+        )
+
+        # Find number of soil and canopy layers
+        soil_layers = config["abiotic"]["soil_layers"]
+        canopy_layers = config["abiotic"]["canopy_layers"]
+
+        LOGGER.info(
+            "Information required to initialise the abiotic model successfully "
+            "extracted."
+        )
+        return cls(data, update_interval, soil_layers, canopy_layers)
+
+    def setup(self) -> None:
+        """Function to set up the abiotic model."""
+
+        setup_variables = simple_regression.setup_simple_regression(
+            layer_roles=self.layer_roles, data=self.data
+        )
+        update_data_object(data=self.data, output_list=setup_variables)
+
+    def spinup(self) -> None:
+        """Placeholder function to spin up the abiotic model."""
+
+    def update(self) -> None:
+        """Placeholder function to update the abiotic model."""
+        # TODO do I need a counter to access the correct time index from the input data?
```

Comment:
Yes, it is the index of the externally generated time series in `data` that I am interested in here. My model does also not care about the actual datetime because it only updates the data object at each update_interval.
We should definitely discuss this again in more detail.